dotnet-roslyn/analyzers/Audacia.CodeAnalysis.Analyzers/README.md

```diff
@@ -764,4 +764,77 @@ dotnet_diagnostic.ACL1014.allowed_words = B2C,365
 ```
 Note the following:
 - These overrides are case-insensitive. For example. an identifier with 'b2c' in its name would not violate this rule.
-- These overrides can contain letters. For example, an identifier with '2' in its name would violate this rule.
\ No newline at end of file
+- These overrides can contain letters. For example, an identifier with '2' in its name would violate this rule.
+
+## ACL1015 - Controller actions does not have ProducesResponseType attribute when return type is TypedResults
+
+<table>
+<tr>
+    <td>Category:</td>
+    <td>Maintainability</td>
+</tr>
+<tr>
+    <td>Audacia coding standard:</td>
+    <td>N/A</td>
+</tr>
+</table>
+
+ACL1015 checks if a controller action does not have `ProducesResponseType` attribute when the return type TypedResults and will produce a warning if not.
+This applies to controller actions defined by either an Http attribute (eg. `HttpGet`) or by the parent class inheriting the `ControllerBase` class.
+
+Code with diagnostic:
+
+```csharp
+[HttpGet]
+[ProducesResponseType(typeof(string), StatusCodes.Status200OK)]
+public public Results<NotFound, Ok<string>> Get()
+{
+    return 'hello';
+}
+```
+
+Code without diagnostic:
+
+```csharp
+[HttpGet]
+public public Results<NotFound, Ok<string>> Get()
+{
+    return 'hello';
+}
+```
+
+## ACL1016 - Controller actions should return TypedResults instead of IActionResults
+
+<table>
+<tr>
+    <td>Category:</td>
+    <td>Maintainability</td>
+</tr>
+<tr>
+    <td>Audacia coding standard:</td>
+    <td>N/A</td>
+</tr>
+</table>
+
+ACL1016 checks if a controller action return type is TypedResults instead of IActionResult and will produce a warning if not.
+This applies to controller actions defined by either an Http attribute (eg. `HttpGet`) or by the parent class inheriting the `ControllerBase` class.
+
+Code with diagnostic:
+
+```csharp
+[HttpGet]
+public public IActionResult Get()
+{
+    return 'hello';
```

Comment:
I don't think the code in either example compiles - would you be able to change it to an example that a developer could mistakenly write & build?